### ISSUE TITLE
V9938/v9958 timing in vblank

### ISSUE THREAD

**Issue by artrag** [openMSX/openMSX#1154]
There is a relevant difference between the tms9918 and the v9938/58 while in vblank.
https://www.msx.org/forum/msx-talk/development/tms-vdp-timing-testing?page=2
On tms you can set the vram for reading and read a byte without any delay 
On Yamaha you need not less than 2 nops between the last out setting the address and the in a, () reading the data.
This happens in vblank and has been tested on 3 real machines 
In openmsx the v9938/58 works like the tms9918 but this is incorrect 
Code in the above link

**Comment by D15C0DE** [comment]
V99x8 does not read VRAM on demand, but it prereads the next VRAM byte in an internal register. When it receives an input request, it will return the value of that register, increment the VRAM address, and preread its value. This means that when you change the data at the address currently set as the read position with a command, the data returned with an IN instruction is the old data. This also means that when you do an input request too fast after setting a new VRAM address, you will receive the data that was at the previously set VRAM address.

How the TMS9918 is able to immediately return the unbuffered data I'm not sure.

**Comment by m9710797** [comment]
@D15C0DE: exactly, and openMSX already (mostly?) implements it like that (including the variable delay for an available CPU-VRAM-access-slot).
What I *think* is still missing is the delay between the start of the VRAM-access-slot and the time the result of the read actually becomes available. Any volunteers to investigate how log that delay is (and to verify that my hypothesis is indeed correct)?

**Comment by artrag** [comment]
In my tests on v99x8 chips you need not less than 2 nops between the last out setting the address and the in a, () reading the data, while on tms9918 you do not need any delay. This in vblank.
